consensus/reactor.go

```diff
@@ -759,21 +768,31 @@ OUTER_LOOP:
 			sleeping = 0
 		}
 
+		// if the proposer is in our peerlist && proposer is the function peer, send the extensions
+		// if the proposer is in our peerlist && proposer is not function peer, do not send the extensions
+		// else if the proposer is not in our peerlist, send extensions to all
+		sendExtensions := true
+		proposerPeerID := p2p.PubKeyToID(rs.Validators.Proposer.PubKey)
+		if conR.Switch.IsPeerUnconditional(proposerPeerID) {
```

Comment:
Why are we checking if the peer is unconditional?

Reply:
https://github.com/skip-mev/cometbft/pull/1/commits/3acd630aa330c2514a11d40a67dfd788662ca31f